lib/Runtime/Language/ModuleNamespace.cpp

```diff
@@ -4,101 +4,133 @@
+        uint localExportCount = 0;
 #endif
-        exportedName->Map([&](PropertyId propertyId) {
-            if (!moduleRecord->ResolveExport(propertyId, nullptr, nullptr, &moduleNameRecord))
-            {
-                // ignore ambigious resolution.
+        if (exportedNames != nullptr)
+        {
+            exportedNames->Map([&](PropertyId propertyId) {
+                if (!moduleRecord->ResolveExport(propertyId, nullptr, nullptr, &moduleNameRecord))
+                {
+                    // ignore ambigious resolution.
 #if DBG
-                unresolvableExportsCount++;
+                    unresolvableExportsCount++;
 #endif
-                return;
-            }
-            // non-ambiguous resolution.
-            if (moduleNameRecord == nullptr)
-            {
-                JavascriptError::ThrowSyntaxError(scriptContext, JSERR_ResolveExportFailed, scriptContext->GetPropertyName(propertyId)->GetBuffer());
-            }
-            if (moduleNameRecord->module == requestModule)
-            {
-                // skip local exports as they are covered in the localExportSlots.
-                return;
-            }
-            Assert(moduleNameRecord->module != moduleRecord);
-            nsObject->AddUnambiguousNonLocalExport(propertyId, moduleNameRecord);
-        });
+                    return;
+                }
+                // non-ambiguous resolution.
+                if (moduleNameRecord == nullptr)
+                {
+                    JavascriptError::ThrowSyntaxError(scriptContext, JSERR_ResolveExportFailed, scriptContext->GetPropertyName(propertyId)->GetBuffer());
+                }
+                if (moduleNameRecord->module == moduleRecord)
+                {
+                    // skip local exports as they are covered in the localExportSlots.
+#if DBG
+                    localExportCount++;
+#endif
+                    return;
+                }
+                Assert(moduleNameRecord->module != moduleRecord);
+                this->AddUnambiguousNonLocalExport(propertyId, moduleNameRecord);
+            });
+        }
 #if DBG
-        uint totalExportCount = (moduleRecord->GetExportedNames(nullptr) != nullptr) ? moduleRecord->GetExportedNames(nullptr)->Count() : 0;
-        uint localExportCount = moduleRecord->GetLocalExportCount();
-        uint unambiguousNonLocalCount = (nsObject->GetUnambiguousNonLocalExports() != nullptr) ? nsObject->GetUnambiguousNonLocalExports()->Count() : 0;
+        uint totalExportCount = exportedNames != nullptr ? exportedNames->Count() : 0;
+        uint unambiguousNonLocalCount = (this->GetUnambiguousNonLocalExports() != nullptr) ? this->GetUnambiguousNonLocalExports()->Count() : 0;
         Assert(totalExportCount == localExportCount + unambiguousNonLocalCount + unresolvableExportsCount);
 #endif
-        BOOL result = nsObject->PreventExtensions();
+        BOOL result = this->PreventExtensions();
         Assert(result);
```

Comment:
This step is not strictly needed as we don't allow property change in the object operation level (we are not really using the dynamictype operations). More a contract visualization :)

---
In reply to: [74163600](https://github.com/Microsoft/ChakraCore/pull/1401#discussion_r74163600) [](ancestors = 74163600)